client/my-sites/plugins/marketplace/marketplace-domain-upsell/index.tsx

```diff
@@ -0,0 +1,171 @@
+/**
+ * External dependencies
+ */
+import React, { useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import DomainPicker from '@automattic/domain-picker';
+import { useShoppingCart } from '@automattic/shopping-cart';
+import CalypsoShoppingCartProvider from 'calypso/my-sites/checkout/calypso-shopping-cart-provider';
+import { domainRegistration } from 'calypso/lib/cart-values/cart-items';
+import { DomainSuggestions } from '@automattic/data-stores';
+import {
+	MARKETPLACE_FLOW_ID,
+	ANALYTICS_UI_LOCATON_MARKETPLACE_DOMAIN_SELECTION,
+} from 'calypso/my-sites/plugins/marketplace/constants';
+import { Button } from '@wordpress/components';
+import { getSelectedSite } from 'calypso/state/ui/selectors';
+
+import './styles.scss';
+import { getProductsList, isProductsListFetching } from 'calypso/state/products-list/selectors';
+
+type DomainSuggestion = DomainSuggestions.DomainSuggestion;
+import { fillInSingleCartItemAttributes } from 'calypso/lib/cart-values';
+
+function Header() {
+	return (
+		<div className="marketplace-domain-upsell__header domains__header">
+			<h1 className="marketplace-domain-upsell__header marketplace-title">Choose a domain</h1>
+			<h2 className="marketplace-domain-upsell__header marketplace-subtitle">
+				Yoast SEO requires a top level domain to index your site and help you rank higher.
+			</h2>
+		</div>
+	);
+}
+
+function DomainPickerContainer( { onDomainSelect, selectedDomain } ) {
+	return (
+		<DomainPicker
+			header={ <Header /> }
+			analyticsUiAlgo={ ANALYTICS_UI_LOCATON_MARKETPLACE_DOMAIN_SELECTION }
+			analyticsFlowId={ MARKETPLACE_FLOW_ID }
+			onDomainSelect={ onDomainSelect }
+			currentDomain={ selectedDomain }
+			hideRecommendationLabel={ true }
+		/>
+	);
+}
+
+function MarketplaceShoppingCart( { onAddDomainToCart, selectedDomain } ) {
```

Comment:
Seems like we have multiple functional components in this file. Does it make sense to move them to separate files?

Reply:
Do we have a convention of using a file per function component @tyxla or any other convention?

Reply:
As per https://github.com/Automattic/wp-calypso/blob/trunk/docs/components.md#glossary:

> A component is represented by the folder holding the main index.jsx file, and a sub-component is any other jsx file used to render pieces of a component. The folder is the "component" namespace and determines the class prefix for any sub-component in that folder. The only folder that matters for namespacing is the immediate container of the jsx files.

Essentially, this best practice applies to the UI components but transcends beyond those and we've been applying it to all the components in Calypso. In my experience, it allows for better readability and reusability.

Reply:
I used to follow this practice which was enforced in the AirBnb styles guide as well. However was influenced against this by this thread https://twitter.com/dan_abramov/status/1220029098543788032. Been defaulting to using multi component files until the code gets unmaintainable. However if its a convention followed frequently I will conform here as well to keep our code base more reliable and readable 👍

Reply:
Yeah, that's also a valid point 😉  I guess it lies somewhere in between: sometimes it's fine to have a small version of a component declared inline next to another one. But also, generally following the practice of one component per file can help reconsider if we really need to split a component into multiple components. Splitting a component into multiple components is often useful, but sometimes we go too far with that, too 😉

Reply:
My take is that if the helper components are not used or usage outside the main component, and if they are not very complex, let's bundle then together in one file. It's kind of a better version of `this.renderSomething()` methods we still have on many class component. And I'm also familiar with Dan Abramov's (frequently expressed) opinions about the one-file-per-component rules 🙂 

This file in question has ~160 lines, which is not a lot.